cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResources.java

```diff
@@ -311,18 +318,26 @@ private void removeKeyVaultAccessPolicyFromDiskEncryptionSetServicePrincipal(Azu
         } else {
             throw new IllegalArgumentException(String.format("Failed to deduce vault resource group name from source vault ID \"%s\"", sourceVaultId));
         }
-        String description = String.format("access to Key Vault \"%s\" in Resource Group \"%s\" for Service Principal having object ID \"%s\" " +
-                        "associated with Disk Encryption Set \"%s\" in Resource Group \"%s\"", vaultName, vaultResourceGroupName, desPrincipalObjectId,
-                desName, desResourceGroupName);
-        retryService.testWith2SecDelayMax15Times(() -> {
-            try {
-                LOGGER.info("Removing {}.", description);
-                azureClient.removeKeyVaultAccessPolicyFromServicePrincipal(vaultResourceGroupName, vaultName, desPrincipalObjectId);
-                LOGGER.info("Removed {}.", description);
-                return true;
-            } catch (Exception e) {
-                throw azureUtils.convertToActionFailedExceptionCausedByCloudConnectorException(e, "Removing " + description);
-            }
-        });
+
+        // Check for the existence of keyVault user has specified before removing disk encryption set's(DES) access permissions from this keyVault.
+        if (!azureClient.checkKeyVaultExistence(vaultResourceGroupName, vaultName)) {
+            LOGGER.info(String.format("vaultName \"%s\" either does not exists/have been deleted or user does not have permissions to access it. " +
+                            "Kindly check if the vault & encryption key exists and correct encryption key url is specified.",
```

Comment:
Nits:
* Possibly use `warn()` log level.
* `vaultName` -> `Vault with name` (though this is not critical, customers will not see this log)
* `url` -> `URL`
* I wonder if the "Kindly check ..." part actually makes sense. The deletion flow will still proceed in this case without attempting the key vault policy updates, and no one will notice this unless explicitly looking at logs. :smile: